src/components/elements/TimePicker.js

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react'
+import TimePicker from 'material-ui/TimePicker'
+import DatePicker from 'material-ui/DatePicker'
+import { FlatButton } from 'material-ui'
+import { DateTime } from 'luxon'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import * as filterActions from 'actions/filterActions'
+
+class TimeAndDatePicker extends Component {
+  static propTypes = {
+    dispatch: PropTypes.func.isRequired,
+    filters: PropTypes.object,
+    history: PropTypes.object.isRequired
+  }
+
+  constructor () {
+    super()
+    this.state = { startDate: null, startTime: null, endDate: null, endTime: null }
```

Comment:
One thing that might be useful as a quality of life improvement would be to set defaults (such as defaulting to the last 24 hours) so that people don't necessarily have to choose all 4 values before submitting.